### PR TITLE
feat: add `sort_by` query param to listing record endpoints

### PR DESCRIPTION
**Pull request by gabrielmbmb**
# Description

This PR adds a `sort_by` query parameter to the `GET /api/v1/datasets/{dataset_id}/records`, `GET /api/v1/me/datasets/{dataset_id}/records` and `POST /api/v1/me/datasets/{dataset_id}/records` that allows to sort the records by using the `inserted_at`, `updated_at` or a metadata property, either in ascending or descending order.

In addition, this PR adds the `parse_query_param` function that is meant to be used as an endpoint dependency using `fastapi.Depends`. The function is able to parse query parameters with the format: `key1,key2,key3` or `key1:value1,value2,value3`.

Closes #<issue_number>

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

**Checklist**

- [ ] I added relevant documentation
- [x] I followed the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [x] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla/server/apis/v1/handlers/datasets.py [hunk 1]
```diff
@@ -91,6 +97,146 @@ async def _get_dataset(
...
+
+async def _build_sort_by(
+    db: "AsyncSession", dataset: Dataset, sort_by_query_param: Optional[Dict[str, str]] = None
+) -> Union[List[SortBy], None]:
+    if sort_by_query_param is None:
+        return None
+
+    sorts_by = []
+    for sort_field, sort_order in sort_by_query_param.items():
+        if sort_field not in ["inserted_at", "updated_at"]:
```
Comment (frascuchon) [inline src/argilla/server/apis/v1/handlers/datasets.py]: Maybe we can define this as constants or `RecordSort` and use them here and also in the `search_engine` module

**Review by frascuchon — APPROVED** [review]
Just a little comment

File: src/argilla/server/apis/v1/handlers/datasets.py (hunk 1 above)
Comment (gabrielmbmb) [reply src/argilla/server/apis/v1/handlers/datasets.py]: yes, agree

File: tests/unit/server/api/v1/test_datasets.py [hunk 2]
```diff
@@ -3756,6 +3959,117 @@ async def test_search_dataset_records_with_wrong_metadata_filter_values(
...
+
+        response = await async_client.post(
+            f"/api/v1/me/datasets/{dataset.id}/records/search",
+            params={"sort_by": "i-do-not-exist:asc"},
+            headers=owner_auth_header,
+            json=query_json,
+        )
+        assert response.status_code == 422
+        assert response.json() == {
+            "detail": f"Provided metadata property in 'sort_by' query param 'i-do-not-exist' not found in dataset with '{dataset.id}'"
```
Comment (frascuchon) [inline tests/unit/server/api/v1/test_datasets.py]:
```suggestion
            "detail": f"Provided metadata property in 'sort_by' query param 'i-do-not-exist' not found in dataset with '{dataset.id}'."
```

File: tests/unit/server/api/v1/test_datasets.py [hunk 3]
```diff
@@ -905,6 +907,104 @@ async def test_list_dataset_records_with_response_status_filter(
...
+        dataset = await DatasetFactory.create()
+
+        response = await async_client.get(
+            f"/api/v1/datasets/{dataset.id}/records",
+            params={"sort_by": "i-do-not-exist:asc"},
+            headers=owner_auth_header,
+        )
+        assert response.status_code == 422
+        assert response.json() == {
+            "detail": f"Provided metadata property in 'sort_by' query param 'i-do-not-exist' not found in dataset with '{dataset.id}'"
```
Comment (frascuchon) [inline tests/unit/server/api/v1/test_datasets.py]:
```suggestion
            "detail": f"Provided metadata property in 'sort_by' query param 'i-do-not-exist' not found in dataset with '{dataset.id}'."
```

File: tests/unit/server/api/v1/test_datasets.py [hunk 4]
```diff
@@ -1371,6 +1472,106 @@ async def test_list_current_user_dataset_records_with_response_status_filter(
...
+        dataset = await DatasetFactory.create()
+
+        response = await async_client.get(
+            f"/api/v1/me/datasets/{dataset.id}/records",
+            params={"sort_by": "i-do-not-exist:asc"},
+            headers=owner_auth_header,
+        )
+        assert response.status_code == 422
+        assert response.json() == {
+            "detail": f"Provided metadata property in 'sort_by' query param 'i-do-not-exist' not found in dataset with '{dataset.id}'"
```
Comment (frascuchon) [inline tests/unit/server/api/v1/test_datasets.py]:
```suggestion
            "detail": f"Provided metadata property in 'sort_by' query param 'i-do-not-exist' not found in dataset with '{dataset.id}'."
```

File: src/argilla/server/apis/v1/handlers/datasets.py [hunk 5]
```diff
@@ -91,6 +97,147 @@ async def _get_dataset(
...
+                raise HTTPException(
+                    status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                    detail=(
+                        f"Provided metadata property in 'sort_by' query param '{sort_field}' not found in dataset with"
+                        f" '{dataset.id}'."
+                    ),
+                )
+            field = metadata_property
+        else:
+            field = sort_field
```
Comment (frascuchon) [inline src/argilla/server/apis/v1/handlers/datasets.py]: I think inverting the if-else logic here, the code will be more readable (long clause in the `else` part):
```python
if sort_field in _RECORD_SORT_FIELD_VALUES:
   field = sort_field
else:
   long code block here
```

**Pull request merged**